# Let $Z_{1}$, $Z_{2}$, ..., $Z_{N}$ describe a branching process in which each parent has $j$ offspring with probability $p_{j}$

# Find the probability $d$ that the process dies out if:

## a) $p_{0} = 1/2$, $p_{1}=p_{2} = 0$, and $p_{3} = 1/2$

## b) $p_{0} = p_{1} = p_{2} = p_{3} = 1/4$

## c) $p_{0} = t$, $p_{1}=1-2t$, $p_{2}=0$ and $p_{3}=t$ where $t\leq 1/2$

____

# a)

# $p_{0} = 1/2$, $p_{1}=p_{2} = 0$, and $p_{3} = 1/2$

# $\implies m = 0\cdot(1/2) + 1\cdot(0) + 2\cdot(0) + 3\cdot(1/2) = 3/2 > 1 \implies d<1$

In [4]:
d = 0.5

for i in range(1000):
    d_new = 0.5+0.5*d**3
    d = d_new
d

0.6180339887498948

# $\implies d\rightarrow 0.618$

____

# b)

# $p_{0} = p_{1} = p_{2} = p_{3} = 1/4$

# $\implies m = 0\cdot(1/4) + 1\cdot(1/4) + 2\cdot(1/4) + 3\cdot(1/4) = 3/2 > 1 \implies d<1$

In [6]:
d = 0.25

for i in range(1000):
    d_new = 0.25*(1+d+d**2+d**3)
    d = d_new
d

0.41421356237309503

# $\implies d\rightarrow 0.414$

_____

# c)

# $p_{0} = t$, $p_{1}=1-2t$, $p_{2}=0$ and $p_{3}=t$ where $t\leq 1/2$

# $\implies m = 0\cdot t + 1\cdot(1-2t)+3\cdot t = 1 + t \geq 1$

# $\implies \text{ if } t=0 \implies d=0$ (since $p_{0}=0$) and otherwise $d<1$

### We can test for a bunch of different possible $t$ values

In [9]:
import numpy as np

In [7]:
def p(j, t):
    if (j==0) or (j==3):
        return t
    elif j==2:
        return 0
    elif j==1:
        return 1-2*t

In [12]:
for t in np.arange(0,0.5,0.05):
    d = t
    for i in range(1000):
        d_new = 0
        for j in range(4):
            d_new += p(j,t)*d**j
        d = d_new
    print(t, d)

(0.0, 0.0)
(0.050000000000000003, 0.6180339887498939)
(0.10000000000000001, 0.61803398874989424)
(0.15000000000000002, 0.61803398874989435)
(0.20000000000000001, 0.61803398874989457)
(0.25, 0.61803398874989457)
(0.30000000000000004, 0.61803398874989468)
(0.35000000000000003, 0.61803398874989468)
(0.40000000000000002, 0.6180339887498949)
(0.45000000000000001, 0.61803398874989479)


# It looks like they all converge to the same $d$ value when $t>0$

## We can derive this value by solving for the roots:

# $d = t + (1-2t)d + td^{3} \implies d = \frac{1}{2}\left (\sqrt{5}-1 \right ) = 0.618034$